functions/index.js

```diff
@@ -0,0 +1,203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import express from 'express'
+
+import { acceptedLanguages, negotiateLanguages } from '@fluent/langneg'
+import AppConstants from '../src/appConstants.js'
+
+import { getSubscribersByHashes } from '../src/db/tables/subscribers.js'
+import { getEmailAddressesByHashes } from '../src/db/tables/emailAddresses.js'
+import { getTemplate } from '../src/views/emails/email2022.js'
+import { breachAlertEmailPartial } from '../src/views/emails/emailBreachAlert.js'
+
+import { bearerToken } from '../src/middleware/util.js'
+
+import {
+  EmailTemplateType,
+  getEmailCtaHref,
+  sendEmail
+} from '../src/utils/email.js'
+import { getMessage } from '../src/utils/fluent.js'
+import {
+  getAddressesAndLanguageForEmail,
+  getBreachByName,
+  loadBreachesIntoApp
+} from '../src/utils/hibp.js'
+import { UnauthorizedError, UserInputError } from '../src/utils/error.js'
+import mozlog from '../src/utils/log.js'
+
+const log = mozlog('controllers.hibp')
+
+const app = express()
+app.use(bearerToken)
+
+/**
+ * Whenever a breach is detected on the HIBP side, HIBP sends a request to this endpoint.
+ * This function attempts to retrieve the breach info from the local cache, if not found
+ * it retrieves it from the database
+ * A breach notification contains the following parameters:
+ * - breachName
+ * - hashPrefix
+ * - hashSuffixes
+ * More about how account identities are anonymized: https://blog.mozilla.org/security/2018/06/25/scanning-breached-accounts-k-anonymity/
+ *
+ * @param {object} req
+ * @param {object} res
```

Comment:
Should this use some data types like `express.Request` and `express.Response` or &lt;other&gt;?

```
git grep "import('express')." | cat

src/controllers/exposureScan.js: * @type {import('express').RequestHandler}
src/controllers/exposures.js: * @type {import('express').RequestHandler}
src/controllers/exposures.js: * @param {NonNullable<import('express').Request['user']>} user
src/controllers/exposures.js: * @returns {user is import('express').Request['user'] & { onerep_profile_id: number }} Whether the user has set up exposure scanning already
src/controllers/requestBreachScan.js:/** @type {import('express').RequestHandler<unknown, RequestBreachScanResponse>} */
src/controllers/storeExposureScanData.js:/** @type {import('express').RequestHandler<import('../external/onerep').ProfileData, StoreExposureScanDataResponse>} */
src/customTypes.d.ts:  fxaProfile: NonNullable<import('express').Request['user']>['fxa_profile_json'];
```

Reply:
Yes it does, thanks! This is based on the existing `hibp` controller, which I want to leave in place until we ship and test and confirm that this is working in production, so I'm trying to minimize the amount of change.

I might just go ahead and update this though, seems like a no-brainer. I'll enable typescript checking for this file too.